src/core/lib/channel/message_size_filter.c

```diff
@@ -155,11 +186,22 @@ static void init_channel_elem(grpc_exec_ctx* exec_ctx,
           grpc_channel_arg_get_integer(&args->channel_args->args[i], options);
     }
   }
+  // Get method config table from channel args.
+  const grpc_arg* channel_arg =
+      grpc_channel_args_find(args->channel_args, GRPC_ARG_SERVICE_CONFIG);
+  if (channel_arg != NULL) {
+    GPR_ASSERT(channel_arg->type == GRPC_ARG_POINTER);
+    chand->method_config_table = grpc_method_config_table_ref(
+        (grpc_method_config_table*)channel_arg->value.pointer.p);
```

Comment:
It's not used for the method config, it's used for the method config **table** -- i.e., the set of all method configs, which is essentially the full set of things in the service config.  (The only exception is the LB policy name, which we could make part of this same arg if we wanted.  In fact, we might wind up doing that as part of the work to replace grpc_resolver_result with grpc_channel_args, as per the TODO in resolver_result.h.)